src/app/producer-shelf/components/producer-shelf/producer-shelf.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, ElementRef, HostListener, OnDestroy, ViewChild } from '@angular/core'
+import { KeyboardBindingService } from '../../abstractions/keyboard-binding.service'
+import { KeyBinding } from '../../models/key-binding'
+import { KeyboardBindingMatcher } from '../../services/keyboard-binding.matcher'
+import { IconButton, IconButtonSize } from '../../../shared/enums/icon-button'
+
+@Component({
+  selector: 'sofie-producer-shelf',
+  templateUrl: './producer-shelf.component.html',
+  styleUrls: ['./producer-shelf.component.scss'],
+})
+export class ProducerShelfComponent implements OnDestroy {
+  public keyBindings: KeyBinding[] = []
+  public pressedKeys: string[] = []
+  public dragOffsetInPixels: number = 0
+  public dragHandleOffsetInPixels?: number
+
+  protected readonly IconButton = IconButton
+  protected readonly IconButtonSize = IconButtonSize
+
+  @ViewChild('dragHandle')
+  private readonly dragHandleElement: ElementRef<HTMLDivElement>
+
+  constructor(
+    private readonly keyboardBindingService: KeyboardBindingService,
+    private readonly keyboardBindingMatcher: KeyboardBindingMatcher
+  ) {
+    // TODO: handle closing subscriptions on teardown
```

Comment:
Isn't this already handled by the `unsubsubscribe` in the `ngOnDestroy`?

Reply:
It was indeed. I have emptied the component for now, however.